tutorials/plot_sleep.py

```diff
@@ -0,0 +1,183 @@
+
+# plot events
+mne.viz.plot_events(
+    events_train, event_id=event_id_train, sfreq=raw_train.info['sfreq'])
+
+##############################################################################
+# Epoching
+
+tmax = 30. - 1. / raw_train.info['sfreq']  # tmax in included
+epochs_train = mne.Epochs(
+    raw_train, events_train, event_id_train,
+    tmin=0., tmax=tmax, baseline=None)
+
+tmax = 30. - 1. / raw_test.info['sfreq']  # tmax in included
+epochs_test = mne.Epochs(
+    raw_test, events_test, event_id_test,
+    tmin=0., tmax=tmax, baseline=None)
+
+print(epochs_train)
+print(epochs_test)
+
+##############################################################################
+# Extract features from EEG: relative power in specific frequency bands
+
+eeg_channels = ["EEG Fpz-Cz", "EEG Pz-Oz"]
+
+# specific frequency bands
+freq_bands = {"delta": [0.5, 4.5],
+              "theta": [4.5, 8.5],
+              "alpha": [8.5, 11.5],
+              "sigma": [11.5, 15.5],
+              "beta": [15.5, 30]}
+
+# extract features from training data
+epochs_train.load_data().pick_channels(eeg_channels)
+z_norm = np.linalg.norm(epochs_train.get_data(), axis=-1)
+
+X_train = np.zeros(
+    (epochs_train.events.shape[0], len(eeg_channels) * len(freq_bands)))
+for idx_band, (band, lims) in enumerate(freq_bands.items()):
+    print(idx_band, band, lims)
+
+    z = np.linalg.norm(
+        epochs_train.copy().filter(lims[0], lims[1]).get_data(),
+        axis=-1) / z_norm
+
+    X_train[
+        :, len(eeg_channels) * idx_band:len(eeg_channels) * (idx_band + 1)] = z
+
+# extract features from testing data
+epochs_test.load_data().pick_channels(eeg_channels)
+z_norm = np.linalg.norm(epochs_test.get_data(), axis=-1)
+
+X_test = np.zeros(
+    (epochs_test.events.shape[0], len(eeg_channels) * len(freq_bands)))
+for idx_band, (band, lims) in enumerate(freq_bands.items()):
+    print(idx_band, band, lims)
+
+    z = np.linalg.norm(
+        epochs_test.copy().filter(lims[0], lims[1]).get_data(),
+        axis=-1) / z_norm
+
+    X_test[
+        :, len(eeg_channels) * idx_band:len(eeg_channels) * (idx_band + 1)] = z
+
+# format annotations
+y_train = events_train[:, 2]
+y_test = events_test[:, 2]
+
+##############################################################################
+# Train a classifier and predict on test recording
+
+clf = RandomForestClassifier().fit(X_train, y_train)
+
+acc = accuracy_score(y_test, clf.predict(X_test))
+print("Accuracy score: {}".format(acc))
+
+
+##############################################################################
+# Plot the power spectrum density (PSD) in each stage
+
+_, ax = plt.subplots()
+
+for stage in zip(epochs_train.event_id):
+    epochs_train[stage].plot_psd(area_mode=None, ax=ax, fmin=0.1, fmax=20.)
+
+prop_cycle = plt.rcParams['axes.prop_cycle']
+colors = prop_cycle.by_key()['color']
+[line.set_color(color) for line, color in zip(ax.get_lines(), colors)]
+plt.legend(list(epochs_train.event_id.keys()))
```

Comment:
start by showing PSDs as it explains why power features can be good predictors. Then you can to the code for classification. I would actually load a second subject here and not before so the beginning is easier to follow.